### DISCUSSION TITLE
How does the `.` "Repeat last change" command work

### DISCUSSION THREAD

**Discussion by palango** [helix-editor/helix#1880] [Q&A]
Coming from vim, I wonder how the dot command is supposed to work.

For example, when having the following text and the cursor at the beginning.
```
word1 word2 word3 word4
```

When I type `wd`, `word1` gets deleted. I expect that when I now press `.`, that `word2` would be deleted as well, but this is not what happens.

**Comment by chrsmth** [comment]
I've been using Helix for about a week, and this behaviour surprised me as well. Also came from vim.

It seems to repeat the last change, but not the last selection. So I think `.` treats your `wd` command as _delete selection_. It repeats the delete, but on your **current** selection and not on a new word selection relative your cursor position. Kakoune has the same behaviour.

I think it would be helpful if the previous selection was repeated. I can see that being difficult if the selection was generated in extend mode (`v`).
